imt/node.go

```diff
@@ -1,21 +1,39 @@
 package imt
 
 import (
+	"encoding/binary"
 	"errors"
 	"math/big"
 )
 
 type Node struct {
 	Key     *big.Int
+	Index   uint64
```

Comment:
The `hash` function should maybe be updated to also include the `Index` in it.

Reply:
This was intentionally left out because this affects the circuits (the circuits would also have to make the same change). Because the key <> index relationship is static (i.e. after inserting a key, that key will always have the same index, and vice versa), we don't need to hash both.